external/Microsoft.SqlTools.Hosting.v2/Utility/CommandOptions.cs

```diff
@@ -144,12 +144,22 @@ public virtual void SetLocale(string locale)
         /// culture-specific messages
         /// </summary>
         /// <param name="locale"></param>
-        internal void LocaleSetter(string locale)
+        internal void LocaleSetter(string locale, Boolean useUserSettings = false)
         {
             // Creating cultureInfo from our given locale
             CultureInfo language = new CultureInfo(locale);
             Locale = locale;
 
+            if (useUserSettings)
+            {
+                // Allow the system set Number Format and Date Format to be preserved when changing the locale.
+                NumberFormatInfo NumberFormat = CultureInfo.CurrentCulture.NumberFormat;
```

Comment:
This is going to override the format even for languages that we support, right? @kburtram is that what we want? I guess it seems potentially reasonable to say that the languages in ADS are purely for localization purposes and globalization settings are always going to be taken from the system but I'm not sure that's the more intuitive.

@smartguest What does VS Code do for numbers displayed?

Reply:
Yes this will override the format for languages that we support, but again users seem to prefer it.

Reply:
Where are you getting your "users seem to prefer it" statement from? I'm slightly worried about making this change and having people get upset because it changed what they were used to (if they had a different setting in their system than ADS).

Maybe not a huge issue, but something to consider when making a change that's potentially going to affect a lot of our users.

Reply:
Karl told me that we should respect the user's region settings and that the current behavior is incorrect in regards to user expectations.

Reply:
Alright, if we get complaints we can look into adding a setting then.